### PR TITLE
[Merged by Bors] - feat(Algebra/Algebra/Subalgebra/Rank): add some results on the ranks of subalgebras

### PR DESCRIPTION
**Pull request by acmepjz**
... and they are corollaries of `rank_mul_rank`.

Since their proof essentially depends on the fact that a non-trivial commutative ring satisfies strong rank condition, we put them into a separate file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
Thanks!
maintainer merge

File: Mathlib/Algebra/Algebra/Subalgebra/Rank.lean [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+open FiniteDimensional
+
+namespace Subalgebra
+
+variable {R S : Type*} [CommRing R] [CommRing S] [Algebra R S]
+
+variable (A B : Subalgebra R S)
+
+theorem rank_sup_eq_rank_left_mul_rank_of_free
+    [Module.Free R A] [Module.Free A (Algebra.adjoin A (B : Set S))] :
```
Comment (riccardobrasca) [inline Mathlib/Algebra/Algebra/Subalgebra/Rank.lean:28]: Can you put some (all?) of this assumptions in a `variable` line?

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Algebra/Subalgebra/Rank.lean (hunk 1 above)
Comment (acmepjz) [reply Mathlib/Algebra/Algebra/Subalgebra/Rank.lean:28]: Let me try. I'm afraid that it will pick up wrong assumptions.

**Pull request closed without merging**